src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -364,6 +371,155 @@ private static bool IsSafeForReordering(BoundExpression expression, RefKind kind
+                {
+                    // Handler conversions are not supported in expression lambdas.
+                    Debug.Assert(!_inExpressionLambda);
+                    var interpolationData = operand.InterpolationData.GetValueOrDefault();
+                    var creation = (BoundObjectCreationExpression)interpolationData.Construction;
+
+                    if (creation.Arguments.Length > (interpolationData.HasTrailingHandlerValidityParameter ? 3 : 2))
+                    {
+                        Debug.Assert(!((BoundConversion)argument).ExplicitCastInCode);
+
+                        // We have an interpolated string handler conversion that needs context from the surrounding arguments. We need to store
+                        // all arguments up to and including the last argument needed by this interpolated string conversion into temps, in order
+                        // to ensure we're keeping lexical ordering of side effects.
+                        ensureTempTrackingSetup(ref temps, ref positionsAssignedToTemp);
+                        Debug.Assert(!positionsAssignedToTemp.IsNull);
+
+                        foreach (var placeholder in interpolationData.ArgumentPlaceholders)
+                        {
+                            // Replace each needed placeholder with a sequence of store and evaluate the temp.
+                            var argIndex = placeholder.ArgumentIndex;
+
+                            BoundLocal local;
+                            switch (argIndex)
+                            {
+                                case BoundInterpolatedStringArgumentPlaceholder.InstanceParameter when receiverStored:
+                                    Debug.Assert(visitedReceiver != null && requiresInstanceReceiver);
+                                    local = (BoundLocal)((BoundSequence)visitedReceiver).Value;
+                                    break;
+
+                                case BoundInterpolatedStringArgumentPlaceholder.InstanceParameter:
+                                    Debug.Assert(visitedReceiver != null && requiresInstanceReceiver);
+                                    local = _factory.StoreToTemp(visitedReceiver, out var store, refKind: visitedReceiver.GetRefKind());
+                                    temps.Add(local.LocalSymbol);
+                                    visitedReceiver = _factory.Sequence(ImmutableArray<LocalSymbol>.Empty, ImmutableArray.Create<BoundExpression>(store), local);
+                                    receiverStored = true;
+                                    break;
+
+                                case >= 0 when positionsAssignedToTemp[argIndex]:
+                                    local = visitedArgumentsBuilder[argIndex] switch
+                                    {
+                                        BoundSequence { Value: BoundLocal l } => l,
+                                        BoundLocal l => l, // Can happen for discard arguments
```

Comment:
nit: I didn't follow how this could happen given the handling in `case >= 0:` below (which always puts a `BoundSequence` with a `BoundLocal` value into the `visitedArgumentsBuilder`). I'm just curious

Reply:
When we assign a discard temp, we mark the position as assigned because that local doesn't need a temp.